app/controllers/ops_controller/settings/upload.rb

```diff
@@ -115,4 +115,27 @@ def logo_dir
     Dir.mkdir(dir) unless dir.exist?
     dir.to_s
   end
+
+  def valid_file?(file, type)
+    ext = file.original_filename.split(".").last.downcase  # Alternatively you can use File.extname
+    return false unless ext == type
+
+    case type
+    when "ico"
+      valid_magic_number = "\x00\x00\x01\x00".force_encoding("ASCII-8BIT")
+    when "png"
+      valid_magic_number = "\x89PNG\r\n\x1A\n".force_encoding("ASCII-8BIT")
+    else
+      return false
+    end
+    File.open(file.tempfile.path, 'rb') do |f|
+    magic_number = begin
+      f.readpartial(valid_magic_number.size)
+    rescue EOFError
+      return false
+    end
+
+    magic_number == valid_magic_number
```

Comment:
Seems my commit didn't properly indent.

Reply:
Actually the `end` on line 139 is too far down as well - it should be below the return false; end after line 136